src/dialog/enemyeditor.cpp

```diff
@@ -0,0 +1,1488 @@
+			w_attributes[9] = DropDownField(&local_guyref.misc10, list_digdoggermisc10);
+			break;
+		case eePATRA:
+			w_attributes[3] = DropDownField(&local_guyref.misc4, list_patramisc4);
+			w_attributes[4] = DropDownField(&local_guyref.misc5, list_patramisc5);
+			w_attributes[9] = DropDownField(&local_guyref.misc10, list_patramisc10);
+			w_attributes[19] = DropDownField(&local_guyref.misc20, list_patramisc20);
+			w_attributes[21] = DropDownField(&local_guyref.misc22, list_patramisc22);
+			w_attributes[24] = DropDownField(&local_guyref.misc25, list_patramisc25);
+			w_attributes[25] = DropDownField(&local_guyref.misc26, list_patramisc26);
+			w_attributes[28] = DropDownField(&local_guyref.misc28, list_patramisc28);
+			break;
+		case eePROJECTILE:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_walkmisc1);
+			break;
+		case eeGHOMA:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_gohmamisc1);
+			break;
+		case eeAQUA:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_aquamisc1);
+			break;
+		case eeMANHAN:
+			w_attributes[1] = DropDownField(&local_guyref.misc2, list_manhandlamisc2);
+			break;
+		case eeLANM:
+			w_attributes[2] = DropDownField(&local_guyref.misc3, list_yesnomisc);
+			break;
+		case eeMOLD:
+			w_attributes[1] = DropDownField(&local_guyref.misc2, list_yesnomisc);
+			break;
+		case eeWIZZ:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_wizzrobemisc1);
+			w_attributes[1] = DropDownField(&local_guyref.misc2, list_wizzrobemisc2);
+			w_attributes[3] = DropDownField(&local_guyref.misc4, list_yesnomisc);
+			break;
+		case eeDONGO:
+			w_attributes[9] = DropDownField(&local_guyref.misc10, list_dodongomisc10);
+			break;
+		case eeKEESE:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_keesemisc1);
+			w_attributes[1] = DropDownField(&local_guyref.misc2, list_keesemisc2);
+			w_attributes[2] = DropDownField(&local_guyref.misc3, list_enemies);
+			break;
+		case eeLEV:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_leevermisc1);
+			break;
+		case eeWALLM:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_noyesmisc);
+			break;
+		case eeTRAP:
+			w_attributes[0] = DropDownField(&local_guyref.misc1, list_trapmisc1);
+			w_attributes[1] = DropDownField(&local_guyref.misc2, list_trapmisc2);
+			break;
+		case eeROCK:
+			w_attributes[9] = DropDownField(&local_guyref.misc10, list_rockmisc10);
+			break;
+		case eeNONE: //(Boss Death Triggers)
+			w_attributes[9] = DropDownField(&local_guyref.misc10, list_yesnomisc);
+			break;
+		default:
+			break;
+	}
+	//
+	auto attributes1_tab = Rows_Columns<2,8>();
+	for (int q = 0; q < 16; q++)
+	{
+		attributes1_tab->add(l_attributes[q]);
+		attributes1_tab->add(w_attributes[q]);
+	}
+	//
+	auto attributes2_tab = Rows_Columns<2, 8>();
+	for (int q = 16; q < 32; q++)
+	{
+		attributes1_tab->add(l_attributes[q]);
+		attributes1_tab->add(w_attributes[q]);
+	}
```

Comment:
(requires the misc->attributes[] refactor)
```suggestion
	auto attributes1_tab = Rows_Columns<3, 8>();
	auto attributes2_tab = Rows_Columns<3, 8>();
	auto attrib_tabs[] = {attributes1_tab, attributes2_tab}
	for (int q = 0; q < 32; q++)
	{
		auto& tab = attrib_tabs[q/16];
		tab->add(l_attributes[q] = Label(fitParent = true, textAlign = 2));
		tab->add(ib_attributes[q] = Button(forceFitH = true, text = "?",
			disabled = true,
			onPressFunc = [&, index]()
			{
				InfoDialog("Attribute Info",h_attribute[index]).show();
			}))
		tab->add(sw_attributes[q] = Switcher(
			tf_attributes[q] = TextField(
				type = GUI::TextField::type::INT_DECIMAL,
				maxLength = 7,
				low = -999999,
				high = 999999,
				val = local_guyref.attributes[q],
				fitParent = true,
				onValChangedFunc = [index](GUI::TextField::type, std::string_view, int32_t val)
				{
					local_guyref.attributes[q] = val;
				}
			),
			ddl_attributes[q] = DropDownList(
				vPadding = 0_px,
				fitParent = true, selectedValue = local_guyref.attributes[q],
				onSelectFunc = [q](int32_t val)
				{
					local_guyref.attributes[q] = val;
				}
			)
			));
	}
```